### PR TITLE
EventFilter-EcalRawToDigi: fix clang warning hides overloaded virtual function

### PR DESCRIPTION
**Pull request by gartung**
by adding using directive and adding parameters to function calls to disambiguate from base class function call. If the function is called without the defaulted parameter it could be interpreted as a call to the base class function.

### CODE REVIEW

**Comment by gartung** [comment]
@Dr15Jones  please review

File: EventFilter/EcalRawToDigi/src/DCCEBEventBlock.cc [hunk 1]
```diff
@@ -164,7 +164,7 @@ void DCCEBEventBlock::unpack(const uint64_t * buffer, size_t numbBytes, unsigned
   
     // Unpack SRP block
     if(srChStatus_ != CH_TIMEOUT &&  srChStatus_ != CH_DISABLED){
-      STATUS = srpBlock_->unpack(&data_,&dwToEnd_);
+      STATUS = srpBlock_->unpack(&data_,&dwToEnd_,SRP_NUMBFLAGS);
```
Comment (Dr15Jones) [inline EventFilter/EcalRawToDigi/src/DCCEBEventBlock.cc]: Why was adding the default value needed? Did it fail to compile otherwise?

Comment (gartung) [inline EventFilter/EcalRawToDigi/src/DCCEBEventBlock.cc]: Yes. With the using statement in the header clang now see two signatures of unpack that take two parameters: one is the base class function and the other is the derived class function with one of three parameters defaulted.

Comment (gartung) [inline EventFilter/EcalRawToDigi/src/DCCEBEventBlock.cc]: I had to do something similar for https://github.com/cms-sw/cmssw/pull/17015

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

For #17080 8728cb807ae0f0de8b466faaa9040b3476e06972

Fixing Clang warnings about overloaded functions in EcalRawToDigi. There should be no change in monitored quantities.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_9_0_X_2016-12-19-1100 show no significant differences, as expected.

**Comment by davidlange6** [comment]
+1

**Pull request merged**